py/picca/data.py

```diff
@@ -63,35 +63,38 @@ class forest(qso):
 
     def __init__(self,ll,fl,iv,thid,ra,dec,zqso,plate,mjd,fid,order):
         qso.__init__(self,thid,ra,dec,zqso,plate,mjd,fid)
-        ## rebin
-
-        bins = ((ll-forest.lmin)/forest.dll+0.5).astype(int)
-        w = bins>=0
-        fl=fl[w]
-        iv =iv[w]
-        bins=bins[w]
-
-        ll = forest.lmin + sp.unique(bins)*forest.dll
-        civ = sp.bincount(bins,weights=iv)
-        w=civ>0
-        civ=civ[bins.min():]
-        cfl = sp.bincount(bins,weights=iv*fl)
-        cfl = cfl[bins.min():]
-        w=civ>0
-        cfl[w]/=civ[w]
-        iv = civ
-        fl = cfl
-
 
         ## cut to specified range
-        w= (ll<forest.lmax) & (ll-sp.log10(1+self.zqso)>forest.lmin_rest) & (ll-sp.log10(1+self.zqso)<forest.lmax_rest)
-        w = w & (iv>0)
-        if w.sum()==0:return
-        
-        ll=ll[w]
-        fl=fl[w]
-        iv=iv[w]
+        bins = sp.floor((ll-forest.lmin)/forest.dll+0.5).astype(int)
+        ll = forest.lmin + bins*forest.dll
+        w = (ll>=forest.lmin)
+        w = w & (ll<forest.lmax)
```

Comment:
here we should make the selection on the bins instead of ll:

    w = (bins >= 0) & (bins < sp.floor((forest.lmax-forest.lmin)/forest.dll+0.5))

Reply:
why is the selection on ll not good ?

Reply:
the difference is on the edge bins. Since forest.lmin is a new bin center, we need to collect all pixels with abs(ll-forest.lmin) < dll/2. If we cut ll > forest.lmin we only get half. The same thing happens near forest.lmax...

Reply:
no because I changed the definition of `ll`: `ll = forest.lmin + bins*forest.dll`

Reply:
Ok, I understand. I think it's correct. Did you do a test with rebinning just to double check?

Reply:
yes and it works with rebin=5